test/e2e/data/e2e_conf.yaml

```diff
@@ -229,6 +229,8 @@ providers:
           - sourcePath: "./infrastructure-aws/generated/cluster-template-peered-remote.yaml"
           - sourcePath: "./infrastructure-aws/generated/cluster-template-internal-elb.yaml"
           - sourcePath: "./infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml"
+          - sourcePath: "./infrastructure-aws/generated/cluster-template-nested-multitenancy-clusterclass.yaml"
+          - sourcePath: "./infrastructure-aws/kustomize_sources/nested-multitenancy-clusterclass/clusterclass-multi-tenancy.yaml"
```

Comment:
We shouldn't need this?

Reply:
Not sure what you mean? I am using new templates so they need to be defined here.

Reply:
make generate-test-flavors automatically generate flavors under generated/ folder. See how the other templates are added:           - sourcePath: "./infrastructure-aws/generated/cluster-template-internal-elb.yaml"

We do not publish full e2e test manifests under kustomize folder, they are generated at runtime.

Reply:
oh, I see what you mean. The 2nd line is for clusterclass definition, which is not generated.

Following quick-start ClusterClass pattern: [cluster-api-provider-aws](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml#L231)

Reply:
When we combine ClusterClasses later, I think we will create a folder that contains only clusterclasses.

Reply:
> Following quick-start ClusterClass pattern: [cluster-api-provider-aws](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml#L231)

This is not really the pattern, might have slipped. 

> When we combine ClusterClasses later, I think we can create a folder that contains only clusterclasses.

Creating a separate folder for ClusterClasses is not related to this. Doesn't it work as is when just the template in the generated folder is added to the config file?

Reply:
I have tried removing the below path and run a test and it didn't work. 
```yaml
          - sourcePath: "./infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml"
```

Error
```bash
failed to get the cluster class template for "quick-start": failed to read "clusterclass-quick-start.yaml" from provider's repository "infrastructure-aws"
```

Thoughts?

Reply:
Auto-generated templates use the folder name.
So in this case template name under generated/ should be: cluster-template-nested-multitenancy-clusterclass.yaml

Reply:
A follow up PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3431